WordPress/Sniffs/WP/PostsPerPageSniff.php

```diff
@@ -68,7 +68,7 @@ public function getGroups() {
 	public function callback( $key, $val, $line, $group ) {
 		$this->posts_per_page = (int) $this->posts_per_page;
 
-		if ( $val > $this->posts_per_page ) {
+		if ( $val > $this->posts_per_page || '-1' === $val ) {
```

Comment:
This is checking a value that is a string, not a negative integer.

Reply:
hi, i checked it, and $val is always converted to string. not sure why or where, didn't look into it deeply.

thanks why the tests pass, either with  `-1` and with `'-1'`

Reply:
the token is parse into a string here:

```php
// this->tokens
23: - // "PHPCS_T_MINUS"
24: 1 // "T_LNUMBER"
25: , // "PHPCS_T_COMMA"
```

```php
// call: WordPress/AbstractArrayAssignmentRestrictionsSniff.php L170
// valStart: 23
// valEnd: 25
$val            = $this->phpcsFile->getTokensAsString( $valStart, ( $valEnd - $valStart ) );
```